intel_owl/tasks.py

```diff
@@ -391,6 +396,61 @@ def send_bi_to_elastic(max_timeout: int = 60, max_objects: int = 10000):
         )[:max_objects].send_to_elastic_as_bi(max_timeout=max_timeout)
 
 
+@shared_task(
+    base=FailureLoggedTask, name="send_plugin_report_to_elastic", soft_time_limit=300
+)
+def send_plugin_report_to_elastic(max_timeout: int = 60, max_objects: int = 10000):
+
+    from api_app.analyzers_manager.models import AnalyzerReport
+    from api_app.connectors_manager.models import ConnectorReport
+    from api_app.ingestors_manager.models import IngestorReport
+    from api_app.models import AbstractReport
+    from api_app.pivots_manager.models import PivotReport
+    from api_app.visualizers_manager.models import VisualizerReport
+
+    def _convert_report_to_elastic_document(_class: AbstractReport) -> List[Dict]:
+        upper_threshold = now().replace(second=0, microsecond=0)
+        lower_threshold = upper_threshold - datetime.timedelta(minutes=5)
```

Comment:
I think it's correct to calculate it inside the task: the alternative is to put it in the beat schedule, but this doesn't work because the function is called once when the schedule is defined and the time range would be the same for all the scheduled tasks. Am i wrong ?

Reply:
no, you right. If I remember correctly, in another occasion we managed this case by calculating the time from an element of the database. In this way there's no way of getting this value wrong because this task would change it only at the time of execution. So, if there are any downtimes, there would be no loss of data. (I am afraid of having the sync misaligned cause we lose some data from time to time. That would make data analysis really bad)

(I would still get the time from "now" first and then, instead of doing minus 5 minutes, I would use as lower_threshold the data got from the database of the last update)

Reply:
I'm not sure, I remember what are you talking about, but I didn't find collections: I found some capped collections used to repeat a task in case of failure, it's similar, but not the same. However I found a way to do it with postgres, I proceed with the merge.